addons/google_calendar/models/calendar.py

```diff
@@ -109,22 +109,29 @@ def _odoo_attendee_commands(self, google_event):
         if google_event.exists(self.env):
             existing_attendees = self.browse(google_event.odoo_id(self.env)).attendee_ids
         attendees_by_emails = {tools.email_normalize(a.email): a for a in existing_attendees}
-        for attendee in google_attendees:
-            email = attendee.get('email')
-
+        partners = self.env['mail.thread']._mail_find_partner_from_emails(emails, records=self, force_create=True, extra_domain=[('type', '!=', 'private')])
+        for attendee in zip(emails, partners, google_attendees):
```

Comment:
The 15.0 fixes fixes both Google and Microsoft. What we're after is a fix for Microsoft. This PR looks like it's only addressing Google?

Reply:
The microsoft part of [15.0-fix](https://github.com/odoo/odoo/pull/82454/files) is not applicable here, because the `microsoft_calendar` module has evolved. In addition, the method in `microsoft_calendar` that is edited in the fix, is identical between 14.0 and 15.0

Reply:
So what I did, is to basically migrate what is in the fix commit you shared. And now I want to test it, already asked Niels if he is able to do so too.

Reply:
OK yes, I see now the Microsoft side has evolved, so that fix is not applicable anymore to it.

> migrate what is in the fix commit you shared

Which commit do you mean here?

> testing by Niels

Will be difficult for him if your change is on the Google side, because BMAir just uses the Microsoft part.

Reply:
this one https://github.com/odoo/odoo/pull/82454/files. 

I will check out `microsoft_calendar` (v14) again, to see if there's something I can do about it. The gist here is that the fix for 15 is not applicable to 14, at least not directly.

Reply:
ok got it, pushing another commit asap

Reply:
OK. I just looked into it myself as well, just going from the original traceback. I'm a bit sad about my bad judgement because I thought this would be an easy backport, but it's not. Anyway, what I've found (you may have already found it yourself as well):

- [message_partner_info_from_emails](https://github.com/OCA/OCB/blob/59cf062c23728ea568b80688db94215a5c73af19/addons/mail/models/mail_thread.py#L1680) has a possibility for partner to be Falsy, and only when force_create=True and some other condition is satisfied, will it create a partner and otherwise it will just add a Falsy value for partner to the dict, probably an empty partner recordset since `.id` still works on it
- [command_attendee](https://github.com/OCA/OCB/blob/59cf062c23728ea568b80688db94215a5c73af19/addons/microsoft_calendar/models/calendar.py#L234) will then contain the Falsy value, eventually leading to the creation error of the attendee

Reply:
The first behavior does not seem changed in newer Odoo versions. Maybe/hopefully they fixed the second behavior to check for falsy partner values before adding the record.